### PR TITLE
fix bugs for clvp model

### PR DESCRIPTION
**Pull request by kaixuanliu**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47127)

Two bugs when I run clvp model using latest transformers:

1. Audio attention mask overwrite the text attention mask (processing_clvp.py)

After the processor refactor, ClvpProcessor uses the generic ProcessorMixin.__call__, which merges sub-processor outputs as {**text_inputs, **processed_audio}. Since ClvpFeatureExtractor also returns an attention_mask (for the raw audio, length ~132k), it overwrote the tokenizer's text attention_mask. The model then computed position_ids = attention_mask.cumsum(-1) - 1, producing indices far beyond max_text_tokens (404):

IndexError: index out of range in self  # text_position_embedding
Fix: when both text and audio are provided, pass return_attention_mask=False to the feature extractor so the tokenizer's attention mask (the one the model actually uses) is preserved. Audio-only calls still return the audio attention mask, so existing behavior/tests are unaffected.

2. fix_speech_decoder_output crashed on short sequences (modeling_clvp.py)

The vectorized rewrite hard-coded filler[-3:] = tensor(decoder_fixing_codes[1:]) (3 elements). Generation is stochastic and occasionally yields sequences shorter than 3 tokens, causing:

RuntimeError: The expanded size of the tensor (1) must match the existing size (3) ...
Fix: clamp the number of trailing fixing codes to min(len(decoder_fixing_codes) - 1, seq_len) and only assign when there is room. For normal-length sequences the behavior is identical to before.

Sample code to reproduce the bug:
```
import torch
import datasets
from transformers import ClvpProcessor, ClvpModelForConditionalGeneration

# Define the text
text = "This is an example text."

# Load audio (sampled at 22050 Hz as required by CLVP)
ds = datasets.load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
ds = ds.cast_column("audio", datasets.Audio(sampling_rate=22050))
sample = ds[0]["audio"]
audio_sample, sr = sample["array"], sample["sampling_rate"]

# Load processor and model
processor = ClvpProcessor.from_pretrained("susnato/clvp_dev")
model = ClvpModelForConditionalGeneration.from_pretrained("susnato/clvp_dev")

# Generate speech candidates and compute similarity
processor_output = processor(raw_speech=audio_sample, sampling_rate=sr, text=text, return_tensors="pt")
torch.manual_seed(0)
outputs = model.generate(**processor_output)

print("Logits per speech:", outputs.logits_per_speech)
print("Logits per text:", outputs.logits_per_text)
print("Speech IDs shape:", outputs.speech_ids.shape)
```

@vasqu @zucchini-nlp pls help review, thx!

### CODE REVIEW

File: src/transformers/models/clvp/processing_clvp.py [hunk 1]
```diff
@@ -29,14 +29,22 @@ def __init__(self, feature_extractor, tokenizer):
...
                 "Using `raw_speech` keyword argument is deprecated when calling ClvpProcessor, instead use `audio`."
             )
         kwargs["audio"] = raw_speech
-        return super().__call__(*args, **kwargs)
+
+        # The CLVP model relies on the *text* attention mask. When both text and audio are provided, prevent the
+        # feature extractor's audio attention mask from overriding the tokenizer's attention mask in the merged output.
+        if kwargs["audio"] is not None and text is not None:
+            audio_kwargs = kwargs.get("audio_kwargs") or {}
+            audio_kwargs.setdefault("return_attention_mask", False)
```
Comment (zucchini-nlp) [inline src/transformers/models/clvp/processing_clvp.py]: hmm so the processor used to not return audio-attn-mask prev?

File: src/transformers/models/clvp/processing_clvp.py [hunk 2]
```diff
@@ -29,14 +29,22 @@ def __init__(self, feature_extractor, tokenizer):
...
+            kwargs["audio_kwargs"] = audio_kwargs
```
Comment (zucchini-nlp) [inline src/transformers/models/clvp/processing_clvp.py]: if audio kwargs are passed a non-nested dict but flat kwargs, we end up losing them all. We'll need to first `merge_kwargs` and then perform necessary popping/overriding

File: src/transformers/models/clvp/modeling_clvp.py [hunk 3]
```diff
@@ -1301,7 +1301,11 @@ def fix_speech_decoder_output(self, speech_ids: torch.LongTensor) -> torch.LongT
         speech_ids = speech_ids.masked_fill(is_stop, decoder_fixing_codes[0])
 
         filler = torch.full((seq_len,), decoder_fixing_codes[0], device=speech_ids.device, dtype=speech_ids.dtype)
-        filler[-3:] = torch.tensor(decoder_fixing_codes[1:], device=speech_ids.device, dtype=speech_ids.dtype)
+        num_fixing_codes = min(len(decoder_fixing_codes) - 1, seq_len)
+        if num_fixing_codes > 0:
+            filler[-num_fixing_codes:] = torch.tensor(
+                decoder_fixing_codes[1:][-num_fixing_codes:], device=speech_ids.device, dtype=speech_ids.dtype
+            )
```
Comment (zucchini-nlp) [inline src/transformers/models/clvp/modeling_clvp.py:1308]: not sure about this - cc @ebezzam @eustlb

**Review by zucchini-nlp — COMMENTED** [review]
And lets add a test  pls

File: src/transformers/models/clvp/processing_clvp.py (hunk 1 above)
Comment (kaixuanliu) [reply src/transformers/models/clvp/processing_clvp.py]: Here is the former implementation:
```
if raw_speech is not None:
    inputs = self.feature_extractor(raw_speech, sampling_rate=sampling_rate, **kwargs)
if text is not None:
    encodings = self.tokenizer(text, **kwargs)

if text is None:
    return inputs                       # audio-only → audio attention_mask
elif raw_speech is None:
    return encodings                    # text-only  → text attention_mask
else:
    inputs["input_ids"] = encodings["input_ids"]
    inputs["attention_mask"] = encodings["attention_mask"]   # <-- overwrite
    return inputs
```
For text+audio input, `attention_mask` will be overwritten by text's attention_mask: https://github.com/huggingface/transformers/blob/v4.56.0/src/transformers/models/clvp/processing_clvp.py#L67C9-L74C26

File: src/transformers/models/clvp/processing_clvp.py (hunk 2 above)
Comment (kaixuanliu) [reply src/transformers/models/clvp/processing_clvp.py]: Well, this is a good point, thx for your advice! Have updated the code

File: src/transformers/models/clvp/processing_clvp.py (hunk 1 above)
Comment (zucchini-nlp) [reply src/transformers/models/clvp/processing_clvp.py]: Ah i see, thanks! Lets wait for audio folks to review the model code :)

File: src/transformers/models/clvp/modeling_clvp.py (hunk 3 above)
Comment (kaixuanliu) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: Could you help review here? Thx! @ebezzam @eustlb

Comment (ebezzam) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: sorry also not familiar with this. @kaixuanliu can you explain why this change is needed?

Comment (kaixuanliu) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: When I run the example code above, sometimes we generate very short length output. If `seq_len` is 1 or 2, the slice filler[-3:] has length 1 or 2, but decoder_fixing_codes[1:] has fixed length 3, causing a shape mismatch. This adjustment trims the fixing codes to the number of available positions.

Comment (kaixuanliu) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: @eustlb Hi, could you pls help review here? Many thx!

Comment (kaixuanliu) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: I have updated the reproduce code(added torch seed) to make the crash happen steadily. pls help review, @zucchini-nlp @ebezzam @eustlb

**Comment by zucchini-nlp** [comment]
😆 

I think we need a small approve from audio folks, the number `3` looks magical to me to begin with but overall the solution makes sense

**Comment by IlyasMoutawwakil** [comment]
@ebezzam @eustlb 🙏

File: src/transformers/models/clvp/modeling_clvp.py (hunk 3 above)
Comment (ebezzam) [reply src/transformers/models/clvp/modeling_clvp.py:1308]: thanks @kaixuanliu for the explanation! as @zucchini-nlp said [here](https://github.com/huggingface/transformers/pull/47127#issuecomment-5180065282) this approach seems better / more defensive than the hardcoded 3

**Review by ebezzam — APPROVED** [review]
Take my approval with a grain of salt, as the method [fix_speech_decoder_output](https://github.com/huggingface/transformers/blob/94f09cfec149050b5355bab7f207ac69e21f1a02/src/transformers/models/clvp/modeling_clvp.py#L1283) (where the fix is done) is quite "hacky" and might have been avoided (although at least has less magic numbers than the [original implementation](https://github.com/huggingface/transformers/blob/94f09cfec149050b5355bab7f207ac69e21f1a02/src/transformers/models/clvp/modeling_clvp.py#L1282) 😅 ). 

In any case, I think this solution is in line with the original intention and more defensive!
